# Sustainability challenge
-----

## Project Overview
This project explores sustainability indicators related to Systembolaget using publicly
available data. The goal is to evaluate environmental and social dimensions of sustainability
through exploratory data analysis and descriptive metrics.


## The brief

> **URGENT: Sustainability team request**
>
> The sustainability team is presenting to leadership Monday morning. They need a quick data snapshot of our product assortment's sustainability profile before EOD.
>
> Use the Systembolaget data to answer their questions below.
>
> **Present your findings clearly** - they're not technical people. Use markdown cells to write your conclusions.
>
> **Deadline: EOD**

---

## Data source & usage
Data used in this analysis was obtained from publicly available sources and is used here
strictly for educational and non-commercial purposes.


## Getting started

Run this cell to load the data:

In [1]:
import requests
import pandas as pd

response = requests.get("https://susbolaget.emrik.org/v1/products")
data = response.json()
df = pd.DataFrame(data)

print(f"Loaded {len(df)} products")

Loaded 27364 products


### Useful columns for this assignment:

| Column | Description |
|--------|-------------|
| `isOrganic` | True/False - is the product organic? |
| `isSustainableChoice` | True/False - Systembolaget's sustainable choice label |
| `isClimateSmartPackaging` | True/False - climate-smart packaging |
| `isEthical` | True/False - ethical certification |
| `categoryLevel1` | Main category (Vin, Öl, Sprit, etc.) |
| `country` | Country of origin |
| `price` | Price in SEK |


<details>
<summary><strong>Click here for full column reference</strong></summary>

### Product identification
| Column | Type | Description |
|--------|------|-------------|
| `productId` | string | Unique product identifier |
| `productNumber` | string | Product number (full) |
| `productNumberShort` | string | Short product number |
| `productNameBold` | string | Product name (brand/main name) |
| `productNameThin` | string | Product name (variant/description) |

### Producer & Supplier
| Column | Type | Description |
|--------|------|-------------|
| `producerName` | string | Name of the producer |
| `supplierName` | string | Name of the supplier |

### Categories
| Column | Type | Description |
|--------|------|-------------|
| `category` | string | General category |
| `categoryLevel1` | string | Main category (Vin, Öl, Sprit, Cider & blanddrycker, Alkoholfritt) |
| `categoryLevel2` | string | Sub-category (e.g., Rött vin, Vitt vin, Lager) |
| `categoryLevel3` | string | More specific category |
| `categoryLevel4` | string | Most specific category |
| `customCategoryTitle` | string | Custom display category |

### Origin
| Column | Type | Description |
|--------|------|-------------|
| `country` | string | Country of origin |
| `originLevel1` | string | Region level 1 |
| `originLevel2` | string | Region level 2 (more specific) |
| `isManufacturingCountry` | bool | Is it manufactured in the listed country? |

### Price & Volume
| Column | Type | Description |
|--------|------|-------------|
| `price` | float | Price in SEK |
| `volume` | int | Volume in ml |
| `volumeText` | string | Volume as text (e.g., "750 ml") |
| `alcoholPercentage` | float | Alcohol percentage |
| `recycleFee` | float | Recycling fee (pant) |

### Sustainability flags
| Column | Type | Description |
|--------|------|-------------|
| `isOrganic` | bool | Organic certification |
| `isSustainableChoice` | bool | Systembolaget's sustainable choice |
| `isClimateSmartPackaging` | bool | Climate-smart packaging |
| `isEthical` | bool | Ethical certification |
| `ethicalLabel` | string | Name of ethical certification |

### Packaging
| Column | Type | Description |
|--------|------|-------------|
| `bottleText` | string | Bottle/container type (Flaska, Burk, etc.) |
| `packagingLevel1` | string | Packaging category |
| `packagingCO2ImpactLevel` | string | CO2 impact level (Lägre, Medel, Högre) |

### Taste characteristics
| Column | Type | Description |
|--------|------|-------------|
| `taste` | string | Taste description (free text) |
| `tasteSymbols` | list | Taste symbols/pairings |
| `tasteClockSweetness` | int | Sweetness (0-12 scale) |
| `tasteClockBody` | int | Body/fullness (0-12 scale) |
| `tasteClockFruitacid` | int | Fruit acidity (0-12 scale) |
| `tasteClockBitter` | int | Bitterness (0-12 scale) |
| `tasteClockRoughness` | int | Roughness/tannins (0-12 scale) |
| `tasteClockSmokiness` | int | Smokiness (0-12 scale) |
| `tasteClockCasque` | int | Oak/barrel character (0-12 scale) |
| `color` | string | Color description |
| `usage` | string | Serving suggestion |

### Wine-specific
| Column | Type | Description |
|--------|------|-------------|
| `vintage` | string | Vintage year |
| `grapes` | list | Grape varieties |
| `seal` | string | Closure type (cork, screw cap, etc.) |

### Stock & Availability
| Column | Type | Description |
|--------|------|-------------|
| `isCompletelyOutOfStock` | bool | Completely out of stock |
| `isTemporaryOutOfStock` | bool | Temporarily out of stock |
| `isDiscontinued` | bool | Product discontinued |
| `assortmentText` | string | Assortment type (Fast, Tillfälligt, etc.) |
| `assortment` | string | Assortment code |

### Launch & News
| Column | Type | Description |
|--------|------|-------------|
| `isNews` | bool | Is it a new product? |
| `isWebLaunch` | bool | Web launch product |
| `productLaunchDate` | string | Launch date (ISO format) |

### Historical data (nested)
| Column | Type | Description |
|--------|------|-------------|
| `priceHistory` | list | List of {x: timestamp, y: price} |
| `alcoholHistory` | list | List of {x: timestamp, y: alcohol%} |
| `soldVolume` | list | List of {x: timestamp, y: volume sold} |

### Other
| Column | Type | Description |
|--------|------|-------------|
| `isKosher` | bool | Kosher certification |
| `sugarContent` | float | Sugar content (g/l) |
| `sugarContentGramPer100ml` | float | Sugar per 100ml |
| `images` | list | Product images |

</details>

## Analytical Approach
The analysis focuses on identifying patterns and gaps in sustainability performance using
quantitative indicators. Results are interpreted in the context of broader sustainability
principles rather than operational decision-making.


---

## 1. The big picture

*What percentage of our assortment is organic? What about sustainable choice and climate-smart packaging? Etc.*

In [15]:
is_organic = df["isOrganic"].value_counts(normalize = True)*100
is_organic

isOrganic
False    87.655314
True     12.344686
Name: proportion, dtype: float64

In [16]:
sustain = df["isSustainableChoice"].value_counts(normalize = True)*100
sustain

isSustainableChoice
False    98.337231
True      1.662769
Name: proportion, dtype: float64

In [17]:
cli_pack = df["isClimateSmartPackaging"].value_counts(normalize = True)*100
cli_pack

isClimateSmartPackaging
False    100.0
Name: proportion, dtype: float64

**Your findings:**

*Write your conclusions here...*

Only 12% of products are organic. If I consider only sustainable choices, the share drops to just 1.66%. Surprisingly, I have not yet identified any climate-smart packaging."

---

## 2. Category breakdown

*Which product categories have the highest percentage of organic products? Which have the lowest? Other intereting findings?*

In [56]:
high_product_per = (df[df["isOrganic"] == True]["categoryLevel1"].value_counts(normalize = True)*100)

In [57]:
high_product_per # product categories have the highest percentage of organic products

categoryLevel1
Vin                     83.925400
Sprit                    8.288928
Öl                       4.618117
Cider & blanddrycker     2.161042
Alkoholfritt             1.006513
Name: proportion, dtype: float64

In [58]:
high_product_per.nsmallest() #lowest percentage of organic products

categoryLevel1
Alkoholfritt             1.006513
Cider & blanddrycker     2.161042
Öl                       4.618117
Sprit                    8.288928
Vin                     83.925400
Name: proportion, dtype: float64

**Your findings:**

*Write your conclusions here...*
After analyzing the data, I found that wine accounts for 83% of the total share in organic products. In contrast, alkoholfritt secures only 1% of the total, making it the lowest category in terms of organic product share.

---

## 3. The price question

*Leadership will ask: "Do organic products cost more?"*

*Compare the average price of organic vs non-organic products. What's the difference? Has it changed over time?*

In [68]:
avg_price_org= df.loc[df["isOrganic"]== True , "price"].mean()

In [69]:
avg_price_org

np.float64(378.7867081113085)

In [71]:
avg_price_not_org = df.loc[df["isOrganic"] == False , "price"].mean()

In [77]:
avg_price_not_org

np.float64(657.6837109147002)

In [72]:
diffrence = avg_price_not_org-avg_price_org

In [73]:
diffrence

np.float64(278.89700280339173)

**Your findings:**
When I compared organic and non-organic products by their average price, I found that non-organic products cost around 657, while organic products average 378. This results in a price difference of 278.


---

## 4. Origin analysis

*Which countries produce the most organic products in our assortment? What origins are popular? Does it differ for categories? And is it different from non-organic products?*

In [86]:
top_coun= (df[df["isOrganic"] == True] ["country"].value_counts().nlargest(5))

In [87]:
top_coun

country
Frankrike    898
Italien      781
Spanien      482
Sverige      435
Österrike    192
Name: count, dtype: int64

In [117]:
#What origins are popular
top_origin= (df[df["isOrganic"] == True]["originLevel1"].value_counts().head(20))

In [118]:
top_origin

originLevel1
Toscana                 202
Bourgogne               158
Rhonedalen              135
Katalonien              119
Piemonte                115
Alsace                  108
Venetien                108
Loiredalen              107
Sicilien                 98
Niederösterreich         90
Languedoc-Roussillon     87
Cava                     80
Rheinhessen              60
Pfalz                    59
Provence                 58
Stockholms län           43
Kastilien-La Mancha      42
Skåne län                42
Västra Götalands län     41
Apulien                  35
Name: count, dtype: int64

In [126]:
top_cate = (df[df["isOrganic"] == True].groupby("categoryLevel1")[["country"]].value_counts().head(20))# Does it differ for categories

In [128]:
top_cate

categoryLevel1        country              
Alkoholfritt          Sverige                   25
                      Frankrike                  2
                      Internationellt märke      2
                      Litauen                    2
                      Danmark                    1
                      Italien                    1
                      Österrike                  1
Cider & blanddrycker  Sverige                   45
                      Frankrike                 13
                      Danmark                    8
                      Estland                    2
                      Storbritannien             2
                      Internationellt märke      1
                      Italien                    1
                      Spanien                    1
Sprit                 Sverige                  180
                      Danmark                   34
                      Frankrike                 17
                      Italien         

In [132]:
#differ from non organic product
top_non_organ= (df[df["isOrganic"] == False]["country"].value_counts().head(20))

In [133]:
top_non_organ

country
Sverige                  5207
Frankrike                4357
Italien                  3669
Storbritannien           1699
Spanien                  1521
USA                      1059
Tyskland                  780
Sydafrika                 698
Portugal                  567
Australien                399
Belgien                   252
Japan                     240
Österrike                 237
Chile                     218
Internationellt märke     204
Mexiko                    190
Nederländerna             178
Danmark                   168
Irland                    165
Nya Zeeland               164
Name: count, dtype: int64

In [136]:
top_non_cate = (df[df["isOrganic"] == False].groupby("categoryLevel1")[["country"]].value_counts().head(50))# Does it differ for categories non organic

In [137]:
top_non_cate

categoryLevel1        country              
Alkoholfritt          Sverige                    52
                      Frankrike                  18
                      Spanien                    14
                      Tyskland                   13
                      Internationellt märke      11
                      Belgien                    10
                      Italien                    10
                      Norge                       6
                      Storbritannien              6
                      Nederländerna               5
                      Sydafrika                   4
                      Australien                  3
                      Danmark                     3
                      Finland                     3
                      Japan                       2
                      Polen                       2
                      Österrike                   2
                      Irland                      1
                    

**Your findings:**
If I considder top 3 countries produce organic product is France, Italy and Spain and non organic is Sweden, France and Italy. 

Top 3 origin label product are Toscana,Bourgogne,Rhonedalen.

When I checked category-wise, it appears that Sweden holds the largest share in almost every category in Organic and Non Organic. 


## 5. Overlap analysis

*How many products are both organic AND sustainable choice? How many are organic OR sustainable choice? How does the overlaps look like for different categories?*

In [122]:
organ_sus = df[(df["isOrganic"]== True) & (df["isSustainableChoice"] == True)]

In [138]:
organ_sus

productId productNumber                    productNameBold  \
152    57754527       1000835                               FORS   
210    38953006       9502601             Château de la Gravelle   
3422    1168752       7373801                         King’s Bay   
3425      15535       1279701                 Castillo de Gredos   
3426      16465       1228401                       Jean Biecher   
...         ...           ...                                ...   
23657  41517590        284801                  Herdade dos Lagos   
23720     10900       1241901  Casal di Serra Classico Superiore   
23723  32063197       1305007             La Sangria de la Playa   
23956  41575749       3207215                         Nils Oscar   
26195   1122715       7401404                   Petit Paul Sapin   

                         productNameThin category productNumberShort  \
152    Tom Collins Organic Non-Alcoholic     None              10008   
210            Vignobles Günther-Chéreau     None              95026   
3422                     Sauvignon Blanc     None              73738   
3425                            Red Wine     None              12797   
3426            Riesling Organic Réserve     None              12284   
...                                  ...      ...                ...   
23657                           Carapeto     None               2848   
23720                       Umani Ronchi     None              12419   
23723                               None     None              13050   
23956                Magro´s ljusa lager     None              32072   
26195                         Merlot ECO     None              74014   

                             producerName                        supplierName  \
152    Lissåker Nord Bjärneman Company AB  Lissåker Nord Bjärneman Company AB   
210             Vignobles Günther-Chéreau                          Vinunic AB   
3422                           Icon Wines             The Wine Team Global AB   
3425                     Bodegas Vinartis                         Bibendum AB   
3426                         Jean Biecher             The Wine Team Global AB   
...                                   ...                                 ...   
23657                   Herdade dos Lagos                        Iconic Wines   
23720                        Umani Ronchi             Enjoy Wine & Spirits AB   
23723                     Anora Group PLC                     Anora Sweden AB   
23956                          Nils Oscar                       Nils Oscar AB   
26195                          Paul Sapin                         Derivino AB   

       isKosher          bottleText  ...  isTsLsAssortment  isTssAssortment  \
152       False                Burk  ...             False            False   
210       False  Lättare glasflaska  ...             False            False   
3422      False  Lättare glasflaska  ...             False            False   
3425      False                Papp  ...             False            False   
3426      False  Lättare glasflaska  ...             False            False   
...         ...                 ...  ...               ...              ...   
23657     False  Lättare glasflaska  ...             False            False   
23720     False  Lättare glasflaska  ...             False            False   
23723     False                 Box  ...             False            False   
23956     False                Burk  ...             False            False   
26195     False          PET-flaska  ...             False            False   

       isTstAssortment  isTsvAssortment  isFsTsAssortment    changedDate  \
152               True            False              True  1757887200000   
210              False             True              True  1759960800000   
3422             False            False              True  1709161200000   
3425             False            False              True  1740697200000   
3426             False

In [124]:
organ_or_sus = df[(df["isOrganic"]== True) | (df["isSustainableChoice"] == True)]

In [125]:
organ_or_sus

productId productNumber            productNameBold  \
0      34724573        196115        Stockholms Bränneri   
5        858241       9600201               Tegnér & Son   
13       857804       9701201               Tegnér & Son   
21       967573       1140903                  Melleruds   
23     14017899       9007101              Alpenglitzern   
...         ...           ...                        ...   
27349  43375233       9554401                 Pie Franco   
27350  24417714       9559301  Chinon Les Petites Roches   
27352  47345789       9568101         Domaine Saint-Remy   
27357  54643979       9580401                   Carchelo   
27363  27913504      63072901                    Reyneke   

                   productNameThin category productNumberShort  \
0      Winter Spritz Non Alcoholic     None               1961   
5      Wilckens Fat Starkvinsglögg     None              96002   
13              Ekologisk Vinglögg     None              97012   
21                     Prima Julöl     None              11409   
23                Organic Glühwein     None              90071   
...                            ...      ...                ...   
27349                Casa Castillo     None              95544   
27350               Charles Joguet     None              95593   
27352                      Eclipse     None              95681   
27357          Monastrell Garnacha     None              95804   
27363   Biodynamic Sauvignon Blanc     None             630729   

                     producerName                supplierName  isKosher  \
0             Stockholms Bränneri      Stockholms Bränneri AB     False   
5        Tegnér Spirits & Wine AB    Tegnér Spirits & Wine AB     False   
13       Tegnér Spirits & Wine AB    Tegnér Spirits & Wine AB     False   
21                      Spendrups       Spendrups Bryggeri AB     False   
23     Andreas Oster Weinkellerei             Great Brands AB     False   
...                           ...                         ...       ...   
27349               Casa Castillo        Wine Group Sweden AB     False   
27350              Charles Joguet            Tryffelsvinet AB     False   
27352          Domaine Saint-Remy  Handpicked Wines Sweden AB     False   
27357            Bodegas Carchelo           Moestue & Cask AB     False   
27363              Reyneke Winery                 Bibendum AB     False   

               bottleText  ...  isTsLsAssortment  isTssAssortment  \
0                    Burk  ...             False             True   
5                  Flaska  ...             False             True   
13                 Flaska  ...             False             True   
21                 Flaska  ...             False             True   
23                 Flaska  ...             False             True   
...                   ...  ...               ...              ...   
27349  Lättare glasflaska  ...             False            False   
27350              Flaska  ...             False            False   
27352  Lättare glasflaska  ...             False            False   
27357              Flaska  ...             False            False   
27363              Flaska  ...             False            False   

       isTstAssortment  isTsvAssortment  isFsTsAssortment    changedDate  \
0                False            False              True  1759442400000   
5                False            False              True  1728597600000   
13               False            False              True  1728597600000   
21               False            False              True  1728597600000   
23               False            False              True  1728597600000   
...                ...              ...               ...            ...   
27349            False            False              True  1712181600000   
27350            False             True              True  1709161200000   
27352            False            False              True  1738882800000   
27

---

## **Sustainability score:** 
Create a new column that counts how many sustainability flags are True for each product (organic, sustainable choice, climate-smart packaging, ethical). Which products score highest?

In [145]:
df["Sustainability_score"] = (
    (df["isOrganic"]== True) & (df["isSustainableChoice"] == True) & (df["isClimateSmartPackaging"] == True) & (df["isEthical"] == True)).astype(int)

In [148]:
df["Sustainability_score"].value_counts()


Sustainability_score
0    27364
Name: count, dtype: int64

---

## Summary for leadership

- Only 12% of products are organic, and sustainable choices alone drop to just 1.66%.

- Wine dominates organic products with 83% share, while alkoholfritt lags at just 1%.

- Non-organic products average 657, compared to 378 for organic, showing a 278 price gap.

- Top organic producers: France, Italy, Spain; Top non-organic: Sweden, France, Italy.

- Sweden consistently holds the largest market share across categories, with 260 products meeting both organic & sustainable criteria versus 3573 meeting only one.

## Limitations
- Reliance on secondary data
- Limited time-series depth
- No access to internal operational metrics